### DISCUSSION TITLE
Force wazuh to perform vulnerability scan

### DISCUSSION THREAD

**Discussion by hexclann** [wazuh/wazuh#14731] [Q&A] [answered]
There is a command to run integrity/rootcheck on all agents  `/var/ossec/bin/agent_control -r -a`. Is there any similar way to force agents to perform vulnerability scanning?

**Comment by branchnetconsulting** [comment]
If you are looking at a given point in time to kick off a fresh vulnerability scan applying to all agents with the purpose of producing a fresh set of Wazuh alerts accounting for all vulnerability detection findings across all agents, you will also have to clear out the existing vulnerability detection state database for each agent (stored on the manager(s)) since new vulnerability detection scans do not generate alerts about findings previously alerted on.  To the best of my knowledge there is no native support for that yet.  Is this your goal?

**Reply by hexclann** [reply]
> If you are looking at a given point in time to kick off a fresh vulnerability scan applying to all agents with the purpose of producing a fresh set of Wazuh alerts accounting for all vulnerability detection findings across all agents

Yes this is exactly what I'm looking for

**Comment by branchnetconsulting** [comment, marked as answer]
I believe Wazuh is moving in the direction of making it possible to get a high level dashboard view of your latest applicable Vulnerability Detection findings across all agents rather than only on a per-agent basis, but I don't know how soon that is going to happen.   In the meantime I'm using a script I wrote that flushes VD state for all agents.  It appears to be working for me but understand it is unofficial beta-quality material.  You could try it (in your dev environment) if you like:

https://github.com/branchnetconsulting/wazuh-tools/blob/master/flush-vd-state

Once you flush VD state for all agents, within less than an hour the VD module on the Wazuh manager(s) should produce a new complete set of VD findings in the form of alerts.  You could then build your own custom dashboard to present them meaningfully in a cross-agent way.

**Reply by hexclann** [reply]
Thank you so much! Previously I was deleting the db but now I can just use your script to perform SQL queries and delete the data
